x/qgb/keeper/keeper_valset.go

```diff
@@ -20,6 +20,9 @@ import (
 // Panics if no valset is found. Because, a valset is always created when starting
 // the chain. Check x/qgb/abci.go:68 for more information.
 func (k Keeper) GetLatestValset(ctx sdk.Context) (*types.Valset, error) {
+	if !k.CheckLatestAttestationNonce(ctx) {
+		return nil, types.ErrLatestAttestationNonceStillNotInitialized
+	}
```

Comment:
If it's just a problem with the querier, then why do all the other functions have this check. Can't we just add the check in the query methods only?

Reply:
Ok, just to clarify where I'm coming from. It's okay if we want to be defensive in our programming but we should also balance that with making the code as easy to read as possible.

Reply:
Makes sense, I guess after we audit this implementation and be sure that no panic would occur, we can get rid of the extra checks. I don't want the QGB  module to cause a network halt at some point or be a pain to deal with while it's still not audited :D